Foundation/NSKeyedArchiver.swift

```diff
@@ -105,13 +105,24 @@ open class NSKeyedArchiver : NSCoder {
         }
     }
     
+    open class func archivedData(withRootObject object: Any, requiringSecureCoding: Bool) throws -> Data {
+        let archiver = NSKeyedArchiver(requiringSecureCoding: requiringSecureCoding)
+        archiver.encode(object, forKey: NSKeyedArchiveRootObjectKey)
+        if let error = archiver.error {
+            throw error
+        } else {
+            return archiver.encodedData
+        }
+    }
+    
     /// Returns an `NSData` object containing the encoded form of the object graph
     /// whose root object is given.
     ///
     /// - Parameter rootObject: The root of the object graph to archive.
     /// - Returns:              An `NSData` object containing the encoded form of the object graph
     ///                         whose root object is rootObject. The format of the archive is
     ///                         `NSPropertyListBinaryFormat_v1_0`.
+    @available(swift, deprecated: 9999, renamed: "archivedData(withRootObject:requiringSecureCoding:)")
     open class func archivedData(withRootObject rootObject: Any) -> Data {
         let data = NSMutableData()
```

Comment:
I could very well be wrong, but just wondering if we could call the new method from here, 

```swift
open class func archivedData(withRootObject rootObject: Any) -> Data {
    archivedData(withRootObject: rootObject: requiringSecureCoding: false)
}
```

Reply:
We can, but in general I prefer not to touch older deprecated methods, so that code only gets new behavior when they adopt new methods.

Reply:
Thanks you. I will follow the same approach in my PRs as well.